torch/csrc/jit/codegen/cuda/root_domain_map.cpp

```diff
@@ -310,9 +310,14 @@ bool ComputeAtRootDomainMap::canMap(
     for (const auto& key_b : getConcretizedKeys(td_b, id_b)) {
       const bool mappable = canMap(key_a, key_b);
       mappable_pair_found = mappable_pair_found || mappable;
-      // If both concrete IDs are not broadcast, they must be mappable
+      // If both concrete IDs are not broadcast, they must be
+      // mappable. Also, if either of the concrete IDs is a reduction,
+      // that means a trivial reduction (i.e., broadcast immediately
+      // followed by reduction), which does not prevent any mapping.
       if (!key_a.concreteId()->isBroadcast() &&
-          !key_b.concreteId()->isBroadcast() && !mappable) {
+          !key_b.concreteId()->isBroadcast() &&
+          !key_a.concreteId()->isReduction() &&
+          !key_b.concreteId()->isReduction() && !mappable) {
```

Comment:
Sorry, I forgot, does td_a and td_b always have a producer-consumer relationship?

Reply:
nevermind, this is grabbing the concrete Id... forget that.